src/renderer/store/battle.store.ts

```diff
@@ -1,16 +1,19 @@
 import { EngineAI, EngineVersion } from "@main/content/engine/engine-version";
 import { GameAI, GameVersion } from "@main/content/game/game-version";
 import { MapData } from "@main/content/maps/map-data";
-import { Battle, BattleWithMetadata, Bot, Faction, GameModeType, isBot, isPlayer, isRaptor, isScavenger, Player, StartPosType } from "@main/game/battle/battle-types";
+import { Battle, BattleWithMetadata, Bot, Faction, GameModeID, GameModeType, isBot, isPlayer, isRaptor, isScavenger, Player, StartPosType, Team } from "@main/game/battle/battle-types";
 import { enginesStore } from "@renderer/store/engine.store";
 import { gameStore } from "@renderer/store/game.store";
 import { getRandomMap } from "@renderer/store/maps.store";
 import { me } from "@renderer/store/me.store";
 import { deepToRaw } from "@renderer/utils/deep-toraw";
+import { spadsBoxToStartBox } from "@renderer/utils/start-boxes";
+import { StartBox } from "tachyon-protocol/types";
 import { reactive, readonly, watch } from "vue";
 
-export const GameMode: Record<string, GameModeType> = {
+export const GameModes: Record<GameModeID, GameModeType> = {
```

Comment:
nit: `GameModes` acts as an enum, and those should be singular, e.g. https://www.typescriptlang.org/docs/handbook/enums.html (and it's not typescript specific)